tycho-bundles/org.eclipse.tycho.core.shared/src/main/java/org/eclipse/tycho/p2/repository/MirroringArtifactProvider.java

```diff
@@ -174,9 +175,12 @@ public final IArtifactDescriptor[] getArtifactDescriptors(IArtifactKey key) thro
             return localArtifactRepository.getArtifactDescriptors(key);
         }
         if (remoteProviders.contains(key)) {
-        	//we can use it if the remote contains it, but want to shadow the repository returned by the descriptor
-        	//just in case someone uses this to download an artifact we are asked again...
+            //we can use it if the remote contains it, but want to shadow the repository returned by the descriptor
+            //just in case someone uses this to download an artifact we are asked again...
             return Arrays.stream(remoteProviders.getArtifactDescriptors(key))
+                    .filter(desc -> Arrays.stream(desc.getProcessingSteps())
```

Comment:
You should be able to use `ArtifactTransferPolicy.isCanonicalFormat(descriptor)`

Reply:
Thanks for the tip. Applied.